forem/forem
# Added tag moderation to the member details view

## What type of PR is this? (check all applicable)


## Description

This introduces tag moderation at the user level in the member details view instead of having to navigate to a specific tag and then add a moderator based on their numeric user id.

cc: @michael-tharrington @ellativity @FounderCasey 

## Related Tickets & Documents

Closes #16522

## QA Instructions, Screenshots, Recordings

1. Run `bin/rails console`
2. In the rails console, enable the admin member view feature flag by running, `FeatureFlag.enable(:admin_member_view)`
3. exit the Rails console.
4. Login as the Forem admin account, email: admin@forem.local, password: password
5. Navigate to http://localhost:3000/admin/users and click on any username in the USER column

![image](https://user-images.githubusercontent.com/833231/153989680-9183b86e-c3ae-4c87-9daa-7c9859980bee.png)

![image](https://user-images.githubusercontent.com/833231/154137791-9c2fdc95-f895-40c6-ba07-d21e1b63fcb4.png)

6. Start typing in the add tags input.

![image](https://user-images.githubusercontent.com/833231/154137115-bc1870b9-c828-473c-9640-54a1d7e444e7.png)

7. Tags should update for the user when saved.


https://user-images.githubusercontent.com/833231/154781522-c710b979-6ca7-4823-9cda-241f96c0cade.mp4

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
